core/domain/opportunity_services.py

```diff
@@ -0,0 +1,370 @@
+        if node is None:
+            continue
+
+        audio_languages_code = [
+            language['id'] for language in constants.SUPPORTED_AUDIO_LANGUAGES]
+        incomplete_translation_languages = audio_languages_code
+        need_voiceartist_in_languages = []
+        if exploration.language_code in incomplete_translation_languages:
+            incomplete_translation_languages.remove(exploration.language_code)
+            need_voiceartist_in_languages.append(exploration.language_code)
+
+        content_count = exploration.get_content_count()
+        translation_counts = exploration.get_translation_counts()
+        complete_translation_lanuages = (
+            exploration.get_languages_with_complete_translation())
+        for language_code in complete_translation_lanuages:
+            incomplete_translation_languages.remove(language_code)
+            need_voiceartist_in_languages.append(language_code)
+
+        exploration_opportunity_summary = (
+            opportunity_domain.ExplorationOpportunitySummary(
+                exp_id, topic.id, topic.name, story.id, story.title, node.title,
+                content_count, copy.deepcopy(incomplete_translation_languages),
+                translation_counts,
+                copy.deepcopy(need_voiceartist_in_languages), []))
+
+        exploration_opportunity_summary_list.append(
+            exploration_opportunity_summary)
+
+    _save_multi_exploration_opportunity_summary(
+        exploration_opportunity_summary_list)
+
+
+def update_exploration_opportunity_with_new_exploration(
+        exp_id, content_count, translation_counts,
+        complete_translation_language_list):
+    """Updates the opportunities models with the changes made in th exploration.
+
+    Args:
+        exp_id: str. The exploration id which is also the id of the opportunity
+            model.
+        content_count: int. The number of content available in the exploration.
+        translation_counts: dict. A dict with language code as a key and
+                number of translation available in that language as a value.
+        complete_translation_language_list: list(str). A list of language code
+            in which the exploration translation is 100%.
+    """
+    model = opportunity_models.ExplorationOpportunitySummaryModel.get(exp_id)
+    exploration_opportunity_summary = (
+        get_exploration_opportunity_summary_from_model(model))
+    exploration_opportunity_summary.content_count = content_count
+    exploration_opportunity_summary.translation_counts = translation_counts
+    exploration_opportunity_summary.complete_translation_lanuages = (
+        complete_translation_language_list)
+
+    new_languages_for_voiceover = set(complete_translation_language_list) - set(
+        exploration_opportunity_summary.assigned_voiceartist_in_languages)
+
+    # We only append new languages to need_voiceartist_in_languages, as the
```

Comment:
Explain in parens what the rejected alternative is. E.g. "We only append new languages to XXX (as opposed to doing YYY), because the .."